### DISCUSSION TITLE
perfview decodes the etl file which contains WPP and trace logging in the same file?

### DISCUSSION THREAD

**Discussion by KieranYu** [microsoft/perfview#2110] [Q&A]
I have an etl file which contains both  WPP and trace logging trace records, so is it possible to use the perfview decode each item no mater it's WPP or tracelogging  type ? 
i also see each trace record will call the EventCallback (https://learn.microsoft.com/zh-cn/windows/win32/api/evntrace/nc-evntrace-pevent_record_callback) , so both WPP and trace logging can be decoded here ?

**Comment by brianrob** [comment]
@KieranYu, I would expect this to work for both types of events in the same file, though I don't know that I've ever explicitly tested this.
